site/zenodo_rdm/legacy/deserializers/schemas.py

```diff
@@ -238,11 +248,203 @@ def _additional_descriptions(self, data):
+            description = legacy_date.get("description")
+            rdm_date = None
+            if start_date and end_date:
+                if start_date == end_date:
+                    rdm_date = start_date
+                else:
+                    rdm_date = f"{start_date}/{end_date}"
+            elif start_date:
+                rdm_date = start_date
+
+            elif end_date:
+                rdm_date = end_date
+            # No start_date nor end_date is not allowed on legacy Zenodo
+
+            rdm_date = {
+                "date": rdm_date,
+                # Type is required on legacy Zenodo
+                "type": {
+                    # Type is a vocabulary on ZenodoRDM.
+                    "id": legacy_date["type"].lower(),
+                },
+            }
+
+            if description:
+                rdm_date.update({"description": description})
+            rdm_dates.append(rdm_date)
+
+        return rdm_dates
+
+    def _references(self, data):
+        """Transform references of a legacy record."""
+        if not data:
+            return None
+
+        rdm_references = []
+
+        for legacy_reference in data:
+            _rdm_reference = {"reference": legacy_reference}
+            rdm_references.append(_rdm_reference)
+
+        return rdm_references
+
+    def _language(self, data):
+        """Transform language of a legacy record."""
+
+        def _map_iso_639_1_to_639_2(language):
+            """Maps ISO 639-1 (2 digits) to ISO 639-2 (3 digits).
+
+            RDM implements 639-3 (3 digits), which is an extension of 639-2.
+            Therefore, mapping from 639-1 to 639-2 should be enough.
+
+            See: https://www.loc.gov/standards/iso639-2/php/code_list.php
+            """
+            mapping = languages_iso_mapping.get(language)
+            # Some 2 digit codes match to multiple 3 digits codes
+            if mapping:
+                return mapping[0]["iso_639_2"]
+            return None
+
+        if not data:
+            return None
+
+        language = None
+
+        # Case 1 - 3 letter code
+        if len(data) == 3:
+            language = data
+        # Case 2  - 2 letter code
+        elif len(data) == 2:
+            language = _map_iso_639_1_to_639_2(data)
+
+        return [{"id": language}] if language else None
+
+    def _related_identifiers(self, data):
+        """Transform related identifiers of a legacy record."""
+        if not data:
+            return None
+
+        related_identifiers = []
+        identifier_schema = IdentifierSchema(allowed_schemes=record_identifiers_schemes)
+        for legacy_identifier in data:
+            # Identifier schema is used to detect the identifier's 'scheme'.
+            # In legacy, 'scheme' is not passed as a parameter. Instead, it's detected from the identifier itself.
+            # TODO schema.load() might error (e.g. invalid identifier scheme). Other fields are not erroring.
```

Comment:
error handling is going to be tackled in a separate [issue](https://github.com/zenodo/zenodo-rdm/issues/334)